drivers/hmis/app/models/hmis/hud/validators/enrollment_validator.rb

```diff
@@ -34,17 +34,25 @@ def self.already_enrolled_full_message
   end
 
   def self.validate_entry_date(enrollment, household_members: nil, options: {})
-    entry_date = enrollment.entry_date
+    entry_date = enrollment&.entry_date
     return [] unless entry_date.present?
 
     errors = HmisErrors::Errors.new
     dob = enrollment.client&.dob
-    exit_date = enrollment&.exit_date
+    exit_date = enrollment.exit_date
 
     errors.add :entry_date, :out_of_range, message: future_message, **options if entry_date.future?
     errors.add :entry_date, :out_of_range, message: over_twenty_years_ago_message, **options if entry_date < (Date.today - 20.years)
     errors.add :entry_date, :out_of_range, message: before_dob_message, **options if dob.present? && dob > entry_date
     errors.add :entry_date, :out_of_range, message: after_exit_message(exit_date), **options if exit_date.present? && exit_date < entry_date
+
+    conflict_scope = enrollment.project
+      .enrollments_including_wip
+      .where(personal_id: enrollment.personal_id, data_source_id: enrollment.data_source_id)
+      .with_conflicting_dates(enrollment.entry_date...)
```

Comment:
I think this is looking for enrollments at the same project.  I'd move the project restriction into the `with_conflicting_dates` `scope e_t[:project_id].eq(project_id)` to enforce that more strongly.

Reply:
+1. It is only used for an individual project, i think it makes sense to make that explicit within the scope.

Reply:
Can do. I prefer to build lean scopes since I find them easier to compose but maybe that leaves too many sharp edges here.